src/transformers/models/fast/configuration_fast.py

```diff
@@ -0,0 +1,182 @@
+# coding=utf-8
+# Copyright The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Fast model configuration"""
+from transformers import PretrainedConfig
+from transformers.utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+FAST_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "fast_base_tt_800_finetune_ic17mlt": (
+        "https://huggingface.co/Raghavan/fast_base_tt_800_finetune_ic17mlt/raw/main/config.json"
+    ),
+}
+
+
+class FastConfig(PretrainedConfig):
+    r"""
+    [Raghavan/fast_base_tt_800_finetune_ic17mlt](https://huggingface.co/Raghavan/fast_base_tt_800_finetune_ic17mlt)
+    """
+
+    def __init__(
+        self,
+        backbone_kernel_size=3,
+        backbone_stride=2,
+        backbone_dilation=1,
+        backbone_groups=1,
+        backbone_bias=False,
+        backbone_has_shuffle=False,
+        backbone_in_channels=3,
+        backbone_out_channels=64,
+        backbone_use_bn=True,
+        backbone_act_func="relu",
+        backbone_dropout_rate=0,
+        backbone_ops_order="weight_bn_act",
+        backbone_stage1_in_channels=[64, 64, 64],
+        backbone_stage1_out_channels=[64, 64, 64],
+        backbone_stage1_kernel_size=[[3, 3], [3, 3], [3, 3]],
+        backbone_stage1_stride=[1, 2, 1],
+        backbone_stage1_dilation=[1, 1, 1],
+        backbone_stage1_groups=[1, 1, 1],
+        backbone_stage2_in_channels=[64, 128, 128, 128],
+        backbone_stage2_out_channels=[128, 128, 128, 128],
+        backbone_stage2_kernel_size=[[3, 3], [1, 3], [3, 3], [3, 1]],
+        backbone_stage2_stride=[2, 1, 1, 1],
+        backbone_stage2_dilation=[1, 1, 1, 1],
+        backbone_stage2_groups=[1, 1, 1, 1],
+        backbone_stage3_in_channels=[128, 256, 256, 256],
+        backbone_stage3_out_channels=[256, 256, 256, 256],
+        backbone_stage3_kernel_size=[[3, 3], [3, 3], [3, 1], [1, 3]],
+        backbone_stage3_stride=[2, 1, 1, 1],
+        backbone_stage3_dilation=[1, 1, 1, 1],
+        backbone_stage3_groups=[1, 1, 1, 1],
+        backbone_stage4_in_channels=[256, 512, 512, 512],
+        backbone_stage4_out_channels=[512, 512, 512, 512],
+        backbone_stage4_kernel_size=[[3, 3], [3, 1], [1, 3], [3, 3]],
+        backbone_stage4_stride=[2, 1, 1, 1],
+        backbone_stage4_dilation=[1, 1, 1, 1],
+        backbone_stage4_groups=[1, 1, 1, 1],
```

Comment:
all of this should be replaced by a single backbone_config, which defines the configuration of the backbone. 

The backbone itself can be added as a standalone model. This allows to then do the following:
```
from transformers import TextNetConfig, FastConfig

backbone_config = TextNetConfig(out_features=["stage1", "stage2", "stage3", "stage4"]
config = FastConfig(backbone_config=backbone_config)

model = FastForSceneTextRecognition(config)
```